### DISCUSSION TITLE
Are data links supported?

### DISCUSSION THREAD

**Discussion by julianorchard** [grafana/grafonnet#219] [Q&A] [answered]
And would it be possible to outline how to add them if they are?

I've found many references to [dashboard links](https://grafana.github.io/grafonnet/API/dashboard/link.html) and [panel links](https://grafana.github.io/grafonnet/API/panel/timeSeries/panelOptions/link.html) but cannot find anything in relation to [data links](https://grafana.com/docs/grafana/latest/panels-visualizations/configure-data-links/). I see that in the old `grafonnet-lib` there was an `addDataLink`/`s` function ([referencing this](https://github.com/grafana/grafonnet-lib/blob/master/grafonnet-7.0/panel/graph.libsonnet#L129)). Is that available here or is there a better method?

**Comment by Duologic** [comment, marked as answer]
https://grafana.github.io/grafonnet/API/panel/timeSeries/index.html#fn-standardoptionswithlinks should give you that.

The items in the array doesn't seem to be defined in the schema, so we don't have a jsonnet counterpart for it. I had a quick look at a local instance and it seems very simple:

```jsonnet
panel
+ standardOptions.withLinks([
  {
    title: 'my title',
    url: 'http://site.com',
    targetBlank: true,
  },
])
```

**Reply by julianorchard** [reply]
Thank you very much!
